### PR TITLE
Lab 04 - Implement numeronym

### PR DESCRIPTION
**Pull request by JackLeeHal**
Fixes #67

#### Changes proposed in this PR:

* Implement the numeronym
* Add test cases

### CODE REVIEW

File: 04_numeronym/jacklee/numeronym.go [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+func main() {
+	fmt.Fprint(out, numeronyms("accessibility", "Kubernetes", "abc"))
+}
+
+func numeronyms(vals ...string) []string {
+	r := make([]string, len(vals))
+	for i, v := range vals {
+		var val = []rune(strings.Trim(v, "\n "))
+		if len(val) < 4 {
```
Comment (mohankrishna) [inline 04_numeronym/jacklee/numeronym.go:20]: `len(val)` is used at multiple places, it is good to assign it to a variable and use the variable where required.

File: 04_numeronym/jacklee/numeronyms_test.go [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+	main()
+
+	// Then
+	r.Equalf(`[a11y K8s abc]`, buf.String(), "Unexpected output in main()")
+}
+
+func TestNumeronyms(t *testing.T) {
+	for _, v := range testData {
+		//When
+		l := v.in
```
Comment (mohankrishna) [inline 04_numeronym/jacklee/numeronyms_test.go:36]: I would suggest getting the actual output from `numeronyms` here and comparing it with the expected value in the next line.

```go
//When
actual := numeronyms(v.in...)

//Then
require.Equal(t, v.expected, actual, "Unexpected output in numeronyms()")
```

File: 04_numeronym/jacklee/numeronym.go [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+var out io.Writer = os.Stdout
+
+func main() {
+	fmt.Fprint(out, numeronyms("accessibility", "Kubernetes", "abc"))
+}
+
+func numeronyms(vals ...string) []string {
+	r := make([]string, len(vals))
+	for i, v := range vals {
+		var val = []rune(strings.Trim(v, "\n "))
```
Comment (mohankrishna) [inline 04_numeronym/jacklee/numeronym.go:19]: You can update this to trim the tab character as well by changing it to

```go 
var val = []rune(strings.Trim(v, "\n\t "))
```
or you can also use `strings.TrimSpace()` which trims \n, \t and spaces

```go 
var val = []rune(strings.TrimSpace(v))
```

https://play.golang.org/p/WlMT1CQepia

**Review by mohankrishna — CHANGES_REQUESTED** [review]
Looks good overall except few small changes.

**Comment by juliaogris** [comment]
Sorry, I will close this PR now in preparation for the next training in a few weeks in Melbourne - feel free to re-open it if you want to continue work on it.

**Pull request closed without merging**